near-sdk/src/environment/mock/mocked_blockchain.rs

```diff
@@ -173,12 +174,16 @@ fn action_to_sdk_action(action: &PrimitivesAction) -> VmAction {
         PrimitivesAction::DeleteAccount(a) => {
             VmAction::DeleteAccount { beneficiary_id: a.beneficiary_id.parse().unwrap() }
         }
+        PrimitivesAction::Delegate(_d) => {
+            panic!("Unimplemented")
+        }
     }
 }
 
 fn pub_key_conversion(key: &VmPublicKey) -> PublicKey {
     // Hack by serializing and deserializing the key. This format should be consistent.
-    String::from(key).parse().unwrap()
+    let key_bytes = [&[key.key_type() as u8], key.key_data()].concat();
+    PublicKey::try_from(key_bytes).unwrap()
```

Comment:
Can you comment on why [`::from_parts`](https://github.com/near/near-sdk-rs/blob/070b024fa93dcf2cf25d32ec570aa8867a8f6ea2/near-sdk/src/types/public_key.rs#LL79C15-L79C15) cannot work here?